src/Mono.Android/Android.Runtime/Logger.cs

```diff
@@ -55,7 +55,7 @@ internal static class Logger {
 						hasNoLibLog = true;
 					}
 				}
-				if (hasNoLibLog)
+				if (AndroidEnvironment.VSAndroidDesignerIsEnabled && hasNoLibLog)
```

Comment:
Is the `VSAndroidDesignerIsEnabled` check slow at all? Should it be second in this `if`-statement?

Reply:
The property just returns the backing field, so it is fast. I prefer `if (AndroidEnvironment.VSAndroidDesignerIsEnabled && hasNoLibLog)` as in release it produces shorter body, linker removes the `&& hasNoLibLog` part of the test.

Like this:
```
...
  catch [System.Private.CoreLib]System.DllNotFoundException 
  {
    IL_0042:  pop
    IL_0043:  ldc.i4.1
    IL_0044:  stsfld     bool Android.Runtime.Logger::hasNoLibLog
    IL_0049:  leave.s    IL_004b
  }  // end handler
  IL_004b:  call       bool Android.Runtime.AndroidEnvironment::get_VSAndroidDesignerIsEnabled()
  IL_0050:  brfalse.s  IL_0052
  IL_0052:  ldloc.1
  IL_0053:  ldc.i4.1
  IL_0054:  add
  IL_0055:  stloc.1
  IL_0056:  ldloc.1
  IL_0057:  ldloc.0
  IL_0058:  ldlen
  IL_0059:  conv.i4
  IL_005a:  blt.s      IL_0023
  IL_005c:  ret
} // end of method Logger::Log
```

In case of `if (hasNoLibLog && AndroidEnvironment.VSAndroidDesignerIsEnabled)` I think the linker has to keep the 1st part of test in place, leading to longer body.

Reply:
Oh, I see it has:
```csharp
internal static bool VSAndroidDesignerIsEnabled { get; } = InitializeVSAndroidDesignerIsEnabled ();
```
And not:
```csharp
internal static bool VSAndroidDesignerIsEnabled => InitializeVSAndroidDesignerIsEnabled ();
```